Formula/nu.rb

```diff
@@ -15,24 +16,40 @@ class Nu < Formula
 
   depends_on "pcre"
 
-  uses_from_macos "llvm" => [:build, :test]
+  uses_from_macos "llvm" => :build
   uses_from_macos "swift" => :build # For libdispatch on Linux.
   uses_from_macos "libffi"
 
   on_linux do
     depends_on "gnustep-make" => :build
     depends_on "gnustep-base"
     depends_on "libobjc2"
+    depends_on "readline"
+  end
+
+  on_arm do
+    # objc/NuBridge.m:1242:6: error: implicit declaration of function 'ffi_prep_closure' is invalid in C99
+    # Since libffi.tbd only exports '_ffi_prep_closure' on x86_64, we need to use formula until fixed.
+    # Issue ref: https://github.com/programming-nu/nu/issues/97
+    depends_on "libffi"
   end
 
   # Clang must be used on Linux because GCC Objective-C support is insufficient.
   fails_with :gcc
 
+  # Fix Snow Leopard or Lion check to avoid `-arch x86_64` being added to ARM build
+  # TODO: Upstream patch
+  patch :DATA
+
   def install
     ENV.delete("SDKROOT") if MacOS.version < :sierra
     ENV["PREFIX"] = prefix
     # Don't hard code path to clang.
     inreplace "tools/nuke", "/usr/bin/clang", ENV.cc
```

Comment:
I don't think this works as intended on macOS since it checks if path exists (https://github.com/programming-nu/nu/blob/v2.3.0/tools/nuke#L280)

It first checks `clang` is a path which fails. Then tries `#{sdk}clang` which fails. Finally, it sets `gcc`.

However, since `gcc` is an alias for `clang` on macOS, there is no actual impact, so is probably okay.